src/status.ts

```diff
@@ -23,123 +23,120 @@ const STARTED_SERVER_OPTIONS = [
   { label: "Ruby LSP: Restart", description: Command.Restart },
 ];
 
-export class StatusItem {
-  private context: vscode.ExtensionContext;
-  private ruby: Ruby;
-  private selector: vscode.DocumentSelector;
-  private serverStatus: vscode.LanguageStatusItem;
-  private yjitStatus: vscode.LanguageStatusItem;
-  private rubyVersionStatus: vscode.LanguageStatusItem;
-  private experimentalFeaturesStatus: vscode.LanguageStatusItem;
-
-  constructor(context: vscode.ExtensionContext, ruby: Ruby) {
-    this.context = context;
-    this.ruby = ruby;
-    this.selector = {
+export abstract class StatusItem {
+  public item: vscode.LanguageStatusItem;
+  protected context: vscode.ExtensionContext;
+
+  constructor(
+    id: string,
+    client: { context: vscode.ExtensionContext; ruby: Ruby }
+  ) {
+    this.item = vscode.languages.createLanguageStatusItem(id, {
       scheme: "file",
       language: "ruby",
+    });
+    this.context = client.context;
+    this.registerCommand();
+  }
+
+  abstract refresh(status: Command, ruby: Ruby): void;
```

Comment:
Do we really need the `ruby` parameter anymore, seen as we already have that being passed to the constructor? I think just a `status` should be enough:
```suggestion
  abstract refresh(status: Command): void;
```